numberformat/proposed.emu

```diff
@@ -24,31 +20,38 @@
         1. Set _intlObj_.[[MinimumIntegerDigits]] to _mnid_.
         1. Let _roundingPriority_ be ? GetOption(_options_, *"roundingPriority"*, *"string"*, &laquo; *"auto"*, *"morePrecision"*, *"lessPrecision"* &raquo;, *"auto"*).
         1. If _mnsd_ is not *undefined* or _mxsd_ is not *undefined*, then
-          1. Set _hasSd_ to *true*.
+          1. Let _hasSd_ be *true*.
         1. Else,
-          1. Set _hasSd_ to *false*.
+          1. Let _hasSd_ be *false*.
         1. If _mnfd_ is not *undefined* or _mxfd_ is not *undefined*, then
-          1. Set _hasFd_ to *true*.
+          1. Let _hasFd_ be *true*.
+        1. Else,
+          1. Let _hasFd_ be *false*.
+        1. If _roundingPriority_ is not *"auto"*, then
```

Comment:
could we change the if to
If _roundingPriority_ is *"auto"*, then
The "If ... is not" pattern is easy to cause double negative and mind twisted later. can we do "if is " not "if is not"?

Reply:
There are 3 possibilities for roundingPriority.  If you don't like "if ... is not ...", I could do "if ... is ... or if ... is ...".  Would that satisfy your suggestion?

Reply:
How about just
```
1. Let _needFd_ be *true*.
1. If _roundingPriority_ is *"auto"*, then
  1. If _hasSd_ is *true*, then
      1. Set _needFd_ to *false*.
  1. Else If _hasFd_ is *false* and _notation_ is *"compact"*, then
      1. Set _needFd_ to *false*.
```